generator/generator_tool/generator_tool.cpp

```diff
@@ -598,26 +599,30 @@ MAIN_WITH_ERROR_HANDLING([](int argc, char ** argv)
 
   string const dataFile = base::JoinPath(path, FLAGS_output + DATA_FILE_EXTENSION);
 
-  if (FLAGS_calc_statistics)
+  if (FLAGS_calc_stats || FLAGS_calc_geom_stats || FLAGS_calc_type_stats)
   {
     LOG(LINFO, ("Calculating statistics for", dataFile));
-
     auto file = OfstreamWithExceptions(genInfo.GetIntermediateFileName(FLAGS_output, STATS_EXTENSION));
-    stats::FileContainerStatistic(file, dataFile);
-
-    stats::MapInfo info;
-    stats::CalcStatistic(dataFile, info);
-    stats::PrintStatistic(file, info);
-  }
-
-  if (FLAGS_type_statistics)
-  {
-    LOG(LINFO, ("Calculating type statistics for", dataFile));
-
     stats::MapInfo info;
-    stats::CalcStatistic(dataFile, info);
-    auto file = OfstreamWithExceptions(genInfo.GetIntermediateFileName(FLAGS_output, STATS_EXTENSION));
-    stats::PrintTypeStatistic(file, info);
+    stats::CalcStatistics(dataFile, info);
+    
+    if (FLAGS_calc_stats)
+    {
+      LOG(LINFO, ("Writing general statistics"));
+      stats::FileContainerStatistics(file, dataFile);
+      stats::PrintStatistics(file, info);
+    }
+    if (FLAGS_calc_geom_stats)
+    {
+      LOG(LINFO, ("Writing geometry statistics"));
+      stats::PrintOuterGeometryStatistics(file, info);
+    }
+    if (FLAGS_calc_type_stats)
+    {
+      LOG(LINFO, ("Writing types statistics"));
```

Comment:
```suggestion
      LOG(LINFO, ("Types statistics for", dataFile));
```

Reply:
What is the point of repeating `dataFile` on each line?
For a `--calc_stats=true --calc_geom_stats=true --calc_type_stats=true` command it'll be printed 4 times!

ATM its clean and logical imho:
```
~/dev/organicmaps$ ../omim-build-release/generator_tool --calc_stats=true --calc_geom_stats=true --calc_type_stats=true --output="Russia_Moscow"
LOG TID(1) INFO    1.699e-05 generator_tool/generator_tool.cpp:656 operator()(): Calculating statistics for ./data/Russia_Moscow.mwm
LOG TID(1) INFO      2.49543 generator_tool/generator_tool.cpp:656 operator()(): Writing general statistics
LOG TID(1) INFO      2.49615 generator_tool/generator_tool.cpp:656 operator()(): Writing geometry statistics
LOG TID(1) INFO      2.49633 generator_tool/generator_tool.cpp:656 operator()(): Writing types statistics
LOG TID(1) INFO      2.49975 generator_tool/generator_tool.cpp:656 operator()(): Stats written to file Russia_Moscow.stats
```

Reply:
The point is to easily grep it from logs if necessary for multi-file processing. But you can leave it as is.